From 3a6f0c2f1219a87ca6f1a3c9f618fcbb5aa215a6 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Sun, 13 Sep 2020 17:31:46 +0300
Subject: [PATCH] [SW-17562] fit: security: invalid key message

Add a message indicating that the provided key is not valid for checking
signature.

Change-Id: I7e493e9c176e01e2ddecb4b1f30381a1e83d1a88
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/preboot/fit.c b/preboot/fit.c
index 8863066..e0e0a42 100644
--- a/preboot/fit.c
+++ b/preboot/fit.c
@@ -315,8 +315,10 @@
 		return 1;
 	}
 
-	if (!preboot_security_key_is_valid(key_idx, key_data, key_len))
+	if (!preboot_security_key_is_valid(key_idx, key_data, key_len)) {
+		preboot_puts("Key validation failed\n");
 		return 1;
+	}
 
 	*pub_key_fit = key_data;
 	return 0;
